core/tests/functional-tests/test_classification.py

```diff
@@ -0,0 +1,1385 @@
+        },
+        {
+            "type": "Recall",
+            "value": 0.0,
+            "label": {"key": "class", "value": "2"},
+        },
+        {"type": "F1", "value": 0.0, "label": {"key": "class", "value": "2"}},
+        {
+            "type": "Precision",
+            "value": 0.5,
+            "label": {"key": "class", "value": "0"},
+        },
+        {
+            "type": "Recall",
+            "value": 1.0,
+            "label": {"key": "class", "value": "0"},
+        },
+        {
+            "type": "F1",
+            "value": 0.6666666666666666,
+            "label": {"key": "class", "value": "0"},
+        },
+    ]
+    for m in eval_job_metrics:
+        if m["type"] not in [
+            "PrecisionRecallCurve",
+            "DetailedPrecisionRecallCurve",
+        ]:
+            assert m in expected_metrics
+    for m in expected_metrics:
+        assert m in eval_job_metrics
+
+    confusion_matrices = eval_job.confusion_matrices
+
+    expected_confusion_matrix = {
+        "label_key": "class",
+        "entries": [
+            {"prediction": "0", "groundtruth": "0", "count": 3},
+            {"prediction": "0", "groundtruth": "1", "count": 3},
+            {"prediction": "1", "groundtruth": "1", "count": 2},
+            {"prediction": "1", "groundtruth": "2", "count": 1},
+            {"prediction": "2", "groundtruth": "1", "count": 1},
+        ],
+    }
+
+    # validate return schema
+    assert confusion_matrices
+    assert len(confusion_matrices) == 1
+    confusion_matrix = confusion_matrices[0]
+    assert "label_key" in confusion_matrix
+    assert "entries" in confusion_matrix
+
+    # validate values
+    assert (
+        confusion_matrix["label_key"] == expected_confusion_matrix["label_key"]
+    )
+    for entry in confusion_matrix["entries"]:
+        assert entry in expected_confusion_matrix["entries"]
+    for entry in expected_confusion_matrix["entries"]:
+        assert entry in confusion_matrix["entries"]
+
+    # validate return schema
+    assert len(confusion_matrices) == 1
+    confusion_matrix = confusion_matrices[0]
+    assert "label_key" in confusion_matrix
+    assert "entries" in confusion_matrix
+
+    # validate values
+    assert (
+        confusion_matrix["label_key"] == expected_confusion_matrix["label_key"]
+    )
+    for entry in confusion_matrix["entries"]:
+        assert entry in expected_confusion_matrix["entries"]
+    for entry in expected_confusion_matrix["entries"]:
+        assert entry in confusion_matrix["entries"]
+
+
+def test_stratify_clf_metrics(
+    gt_clfs_tabular: list[int],
+    pred_clfs_tabular: list[list[float]],
+):
+    assert len(gt_clfs_tabular) == len(pred_clfs_tabular)
+
+    groundtruths = []
+    predictions = []
+    for i, label_value in enumerate(gt_clfs_tabular):
+        if (
+            i % 3 == 2
+        ):  # core doesn't use filters, so this emulates the results of the original test
```

Comment:
nit: We can just remove this test since its filter related.

Reply:
deleted